File: deepest leaves sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
int dep=0,sum=0;
void dfs(TreeNode* root,int curDep){
    if(root==nullptr){return ;}
    if(curDep==dep){sum+=root->val;}
    if(curDep>dep){sum=root->val;dep=curDep;}
    if(root->left){
        dfs(root->left,curDep+1);
    }
    if(root->right){
        dfs(root->right,curDep+1);
    }
}
    int deepestLeavesSum(TreeNode* root) {
        int curDep=0;
        dfs(root,curDep);
        return sum;
    }
};

File: deepest leaves sum.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int sum=0,dep=0;
    void dfs(TreeNode root,int curDep){
        if(root==null){return ;}
        if(curDep==dep){sum+=root.val;}
        if(curDep>dep){dep=curDep;sum=root.val;}
        if(root.left!=null){dfs(root.left,curDep+1);}
        if(root.right!=null){dfs(root.right,curDep+1);}
    }
    public int deepestLeavesSum(TreeNode root) {
        int curDep=0;
        dfs(root,curDep);
        return sum;
    }
}